### PR TITLE
Support of AsciiMath

### PR DESCRIPTION
**Pull request by jackycute**
See #495

### CODE REVIEW

**Comment by ccoenen** [comment]
I did not yet test this on any server, but as far as I can see this is limited to a 2.7.0 -> 2.7.2 upgrade. Does this indeed add AsciiMath support?

**Comment by jackycute** [comment]
Hi @ccoenen 
I've tested on my local build.

Not just upgrade 2.7.0 to 2.7.2.
There are different includes of the config file `TeX-AMS-MML_HTMLorMML` to `TeX-MML-AM_HTMLorMML` which add support of AsciiMath.

Could you help to proof this too?

**Comment by jackycute** [comment]
See more here: http://docs.mathjax.org/en/latest/config-files.html#the-tex-mml-am-htmlormml-configuration-file

**Comment by SISheogorath** [comment]
I think there is some config missing:

https://github.com/hackmdio/hackmd/blob/aca5490b3a237dca88d8607806e89f86aad4c978/public/js/mathjax-config-extra.js#L1-L8

https://docs.mathjax.org/en/latest/options/preprocessors/asciimath2jax.html#configure-asciimath2jax

**Comment by jackycute** [comment]
Thanks @SISheogorath 
But still can't be very sure about is this completely work or not.
Could anyone help?

File: public/js/mathjax-config-extra.js [hunk 1]
```diff
@@ -4,5 +4,8 @@ window.MathJax = {
   tex2jax: {
     inlineMath: [['$', '$'], ['\\(', '\\)']],
     processEscapes: true
+  },
+  asciimath2jax: {
+    delimiters: [['$', '$'], ['\\(', '\\)']]
```
Comment (SISheogorath) [inline public/js/mathjax-config-extra.js]: I'm not sure they can be the same as tex2jax :/

I just checked the official default config and they are different there.
https://github.com/mathjax/MathJax/blob/master/config/default.js#L392-L398

By the way, my test is this:

```
sum_(i=1)^n i^3=((n(n+1))/2)^2
```

From http://asciimath.org/ which also recommends mathjax

**Comment by jackycute** [comment]
Nah, I was wrong.
This can't support using the same syntax with TeX, they have different typesetting.
We could learn from draw.io for using different surround character to mixed them.
https://support.draw.io/questions/2949135/how-to-use-mathematical-typesetting

**Comment by jackycute** [comment]
Since the <code>\`</code> actually overlap the code syntax of Markdown.
I would propose a new syntax that blend them in.
Use <code>\`</code> inside <code>$</code> would make Math using AsciiMath typesetting.
Like:
```
$`sum_(i=1)^n i^3=((n(n+1))/2)^2`$
```
Or
```
$$
`sum_(i=1)^n i^3=((n(n+1))/2)^2`
$$
```

Any ideas?

**Comment by SISheogorath** [comment]
I'm not sure. maybe @ccoenen has an idea for this.

I could also imagine using something like `%%` because \` is definitely reserved from Markdown code blocks.

**Comment by SISheogorath** [comment]
I think if we want to add ASCIImath support we should use `§§` which is uncommon enough to not causing issues and close enough to `$$`.

**Comment by ccoenen** [comment]
Sorry, I have nothing of substance to contribute, I'm afraid.

**Comment by jackycute** [comment]
We need new syntax or yaml metadata option to support AsciiMath.
It's ok to close this for now.

**Comment by cben** [comment]
FWIW ``$`...`$`` is the delimiter for inline LaTeX math in GitLab markdown.

I'm not sure there is any syntax prior art for supporting both LaTeX and AsciiMath in markdown.

**Comment by jackycute** [comment]
I don't think this will land in any close future since it's related to the syntax.
Close for now.

**Pull request closed without merging**